src/server/location/location.service.ts

```diff
@@ -140,4 +145,25 @@ export class LocationService {
         .filter((title) => !!title),
     }));
   }
+
+  /**
+   * Resolves with a list of rooms along with their associated campus and
+   * building information. This is used to populate the Room Admin table.
+   */
+  public async getAdminRooms(): Promise<RoomAdminResponse[]> {
```

Comment:
Thanks Robert! It would definitely be ideal to have fewer service methods, but maybe this could be something that we revisit in the future. I ended up renaming `getRooms` to `getRoomAvailability` and `getAdminRooms` to `getFullRoomList` after talking to Jonathan about this. I also opened the issue #574 so that we have a note to revisit this as a potential enhancement.

Reply:
Okay, I was doing some exploration about the feasibility of this earlier. I'll attach the diff to the issue, thanks for opening that!

Reply:
Thanks for opening that. I guess my concern is here that we seem to keep adding to this. Maybe we can chat about prioritizing fixing #574 in the next sprint planning session. 

I was doing some experimentation earlier about how feasible a change this would be and have a diff that more or less covers it. I'll attach that to #574.

Reply:
@rmainseas Thanks for looking into it, Robert! I took a look at the changes briefly. Do the changes that you added still allow the room combobox in the `Room Schedule` to work as expected? I would expect to not be able to easily replace `getRoomList` with `getFullRoomList` since they have different response DTO shapes?

Reply:
Yeah, if you look at the diff it swaps out the DTO as well to use `RoomResponse` and then updates `RoomResponse` to have the building object. There is a test still failing with that because it expects `$buildingName $roomName`, whereas with these changes, it was just getting `$roomName`. So 'Maxwell Dworkin G20' was showing up as just 'G20'. 

One solution to this would be to tweak the test to look for `room.building.name` instead of just `room.name`, another would be to tweak the client side code similarly. Either way, it would just involve changing which property was being accessed.